src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests_WithElement_Extra.cs

```diff
@@ -7649,4 +7812,102 @@ static void Main()
+                {
+                    _builder.AppendLine("i:" + i.ToString());
+                    _builder.AppendLine("s:" + s);
+                }
+            }
+            """;
+
+        var executableCode = """
+            class Program
+            {
+                static void Main()
+                {
+                    int i = 10;
+                    string s = "arg";
+                    C c = [with(i, s, $"" + $"literal")];
+                }
+            }
+            """;
+
+        var handler = GetInterpolatedStringCustomHandlerType("CustomHandler", "partial struct", useBoolReturns: true);
+
+        CompileAndVerify([code, executableCode, InterpolatedStringHandlerArgumentAttribute, handler], expectedOutput: """
+                i:10
+                s:arg
+                literal:literal
+                """)
+            .VerifyDiagnostics()
+            .VerifyIL("Program.Main", """
+                {
+                  // Code size       45 (0x2d)
+                  .maxstack  7
+                  .locals init (string V_0, //s
+                                int V_1,
+                                string V_2,
+                                CustomHandler V_3)
+                  IL_0000:  ldc.i4.s   10
+                  IL_0002:  ldstr      "arg"
+                  IL_0007:  stloc.0
+                  IL_0008:  stloc.1
+                  IL_0009:  ldloc.1
+                  IL_000a:  ldloc.0
+                  IL_000b:  stloc.2
+                  IL_000c:  ldloc.2
+                  IL_000d:  ldloca.s   V_3
+                  IL_000f:  ldc.i4.7
+                  IL_0010:  ldc.i4.0
+                  IL_0011:  ldloc.1
+                  IL_0012:  ldloc.2
+                  IL_0013:  call       "CustomHandler..ctor(int, int, int, string)"
+                  IL_0018:  ldloca.s   V_3
+                  IL_001a:  ldstr      "literal"
+                  IL_001f:  call       "bool CustomHandler.AppendLiteral(string)"
+                  IL_0024:  pop
+                  IL_0025:  ldloc.3
+                  IL_0026:  newobj     "C..ctor(int, string, CustomHandler)"
+                  IL_002b:  pop
+                  IL_002c:  ret
+                }
+                """);
+    }
+
+    [Fact]
+    public void WithOutsideCollectionIsAnInvocation()
+    {
+        var source = """
+            using System.Collections.Generic;
+            class C
+            {
+                void M()
+                {
+                    N(with(capacity: 0), 1, 2, 3);
+                }
+
+                void N(params List<int> list) { }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics(
+            // (6,11): error CS0103: The name 'with' does not exist in the current context
+            //         N(with(capacity: 0), 1, 2, 3);
+            Diagnostic(ErrorCode.ERR_NameNotInContext, "with").WithArguments("with").WithLocation(6, 11));
+    }
```

Comment:
nit: to be thorough, I'd also include a test for `GetTypeInfo` on the `with` (expected to find/return no information)